%% Generated by Sphinx.
\def\sphinxdocclass{article}
\documentclass[letterpaper,10pt,english]{sphinxhowto}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax
\ifdefined\pdfimageresolution
    \pdfimageresolution= \numexpr \dimexpr1in\relax/\sphinxpxdimen\relax
\fi
%% let collapsible pdf bookmarks panel have high depth per default
\PassOptionsToPackage{bookmarksdepth=5}{hyperref}

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{tgtermes}
\usepackage{tgheros}
\renewcommand{\ttdefault}{txtt}



\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=auto}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}


\usepackage{sphinxmessages}
\setcounter{tocdepth}{2}



\title{VCFHunter}
\date{Jul 18, 2022}
\release{1.0}
\author{Michael Hall}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Version 1.0}
\makeindex
\begin{document}

\pagestyle{empty}

        \pagenumbering{Roman} %%% to avoid page 1 conflict with actual page
        \begin{titlepage}
            \vspace*{10mm} %%% * is used to give space from top
            \flushright\textbf{\Huge {VCFHunter\\}}
            \vspace{0mm} %%% * is used to give space from top
            \textbf{\Large {A Step-by-Step Guide\\}}
            \vspace{0mm} %%% * is used to give space from top
            \textbf{\Large {[DRAFT]\\}}
            \vspace{50mm}
            \textbf{\Large {Michael Hall\\}}
            \vspace{10mm}
            \textbf{\Large {Plant Breeding and Genetics Laboratory\\}}
            \vspace{0mm}
            \textbf{\Large {FAO/IAEA Joint Division\\}}
            \vspace{0mm}
            \textbf{\Large {Seibersdorf, Austria\\}}
	    \vspace{10mm}
            \normalsize Created: July, 2022\\
            \vspace*{0mm}
            \normalsize  Last updated: 14 July 2022
            %% \vfill adds at the bottom
            \vfill
            \small\flushleft {{\textbf {Please note:}} \textit {This is not an official IAEA publication but is made available as working material. The material has not undergone an official review by the IAEA. The views
expressed do not necessarily reflect those of the International Atomic Energy Agency or its Member States and remain the responsibility of the contributors. The use of particular designations of countries or territories does not imply any judgement by the publisher, the IAEA, as to the legal status of such countries or territories, of their authorities and institutions or of the delimitation of their boundaries. The mention of names of specific companies or products (whether or not indicated as registered) does not imply any intention to infringe proprietary rights, nor should it be construed as an endorsement or recommendation on the part of the IAEA.}}
        \end{titlepage}
        \pagenumbering{arabic}
        \newcommand{\sectionbreak}{\clearpage}

\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}



\section{VCFHunter on Sorghum Semi\sphinxhyphen{}Dwarfism}
\label{\detokenize{VCFHunter:vcfhunter-on-sorghum-semi-dwarfism}}\label{\detokenize{VCFHunter::doc}}

\bigskip\hrule\bigskip



\subsection{Software Prerequisites}
\label{\detokenize{VCFHunter:software-prerequisites}}
\sphinxAtStartPar
\#Download git repository:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
git clone https://github.com/SouthGreenPlatform/VcfHunter.git
\PYG{n+nb}{cd} VcfHunter
\PYG{n+nb}{cd} bin
ls

\PYG{c+c1}{\PYGZsh{}See all available python scripts}

VCFtools \PYG{o}{(}\PYG{l+m}{0}.1.15\PYG{o}{)}
Â© Adam Auton and Anthony Marcketta \PYG{l+m}{2009}

Process Variant Call Format files

For a list of options, please go to:
https://vcftools.github.io/man\PYGZus{}latest.html

Alternatively, a man page is available, type:
man vcftools

Questions, comments, and suggestions should be emailed to:
vcftools\PYGZhy{}help@lists.sourceforge.net
\end{sphinxVerbatim}


\subsection{Extract Sample names from VCF}
\label{\detokenize{VCFHunter:extract-sample-names-from-vcf}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
head \PYGZhy{}n \PYG{l+m}{1000} DNAseq\PYGZus{}prefiltered.vcf \PYG{p}{|} grep \PYG{l+s+s2}{\PYGZdq{}\PYGZsh{}CHROM\PYGZdq{}} \PYG{p}{|} sed \PYG{l+s+s1}{\PYGZsq{}s/\PYGZbs{}t/\PYGZbs{}n/g\PYGZsq{}} \PYG{p}{|} tail \PYGZhy{}n +10 \PYGZgt{} all\PYGZus{}names.tab
\end{sphinxVerbatim}


\subsection{Filter VCF with python script}
\label{\detokenize{VCFHunter:filter-vcf-with-python-script}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{}Filter VCF}
python vcfFilter.1.0.py \PYGZhy{}\PYGZhy{}vcf ../data/Sorghumvcf/freebayes\PYGZus{}D2.filtered.vcf \PYGZhy{}\PYGZhy{}names sorghum\PYGZus{}all\PYGZus{}names.tab
\PYGZhy{}\PYGZhy{}MinCov \PYG{l+m}{10} \PYGZhy{}\PYGZhy{}MaxCov \PYG{l+m}{300} \PYGZhy{}\PYGZhy{}MinAl \PYG{l+m}{3} \PYGZhy{}\PYGZhy{}nMiss \PYG{l+m}{1} \PYGZhy{}\PYGZhy{}RmAlAlt \PYG{l+m}{1}:3:4:5:6 \PYGZhy{}\PYGZhy{}prefix DNAseq\PYGZus{}Filtered \PYGZhy{}g y
\end{sphinxVerbatim}


\subsection{Separate VCF by chromomsome using vcftools}
\label{\detokenize{VCFHunter:separate-vcf-by-chromomsome-using-vcftools}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{}Separate VCF by Chromosome}
vcftools \PYGZhy{}\PYGZhy{}gzvcf DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz \PYGZhy{}\PYGZhy{}chr Chr01 \PYGZhy{}\PYGZhy{}recode \PYGZhy{}\PYGZhy{}out ../data/Sorghumvcf/Chr01\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz
vcftools \PYGZhy{}\PYGZhy{}gzvcf DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz \PYGZhy{}\PYGZhy{}chr Chr02 \PYGZhy{}\PYGZhy{}recode \PYGZhy{}\PYGZhy{}out ../data/Sorghumvcf/Chr02\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz
vcftools \PYGZhy{}\PYGZhy{}gzvcf DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz \PYGZhy{}\PYGZhy{}chr Chr03 \PYGZhy{}\PYGZhy{}recode \PYGZhy{}\PYGZhy{}out ../data/Sorghumvcf/Chr03\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz
vcftools \PYGZhy{}\PYGZhy{}gzvcf DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz \PYGZhy{}\PYGZhy{}chr Chr04 \PYGZhy{}\PYGZhy{}recode \PYGZhy{}\PYGZhy{}out ../data/Sorghumvcf/Chr04\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz
vcftools \PYGZhy{}\PYGZhy{}gzvcf DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz \PYGZhy{}\PYGZhy{}chr Chr05 \PYGZhy{}\PYGZhy{}recode \PYGZhy{}\PYGZhy{}out ../data/Sorghumvcf/Chr05\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz
vcftools \PYGZhy{}\PYGZhy{}gzvcf DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz \PYGZhy{}\PYGZhy{}chr Chr06 \PYGZhy{}\PYGZhy{}recode \PYGZhy{}\PYGZhy{}out ../data/Sorghumvcf/Chr06\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz
vcftools \PYGZhy{}\PYGZhy{}gzvcf DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz \PYGZhy{}\PYGZhy{}chr Chr07 \PYGZhy{}\PYGZhy{}recode \PYGZhy{}\PYGZhy{}out ../data/Sorghumvcf/Chr07\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz
vcftools \PYGZhy{}\PYGZhy{}gzvcf DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz \PYGZhy{}\PYGZhy{}chr Chr08 \PYGZhy{}\PYGZhy{}recode \PYGZhy{}\PYGZhy{}out ../data/Sorghumvcf/Chr08\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz
vcftools \PYGZhy{}\PYGZhy{}gzvcf DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz \PYGZhy{}\PYGZhy{}chr Chr09 \PYGZhy{}\PYGZhy{}recode \PYGZhy{}\PYGZhy{}out ../data/Sorghumvcf/Chr09\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz
vcftools \PYGZhy{}\PYGZhy{}gzvcf DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz \PYGZhy{}\PYGZhy{}chr Chr10 \PYGZhy{}\PYGZhy{}recode \PYGZhy{}\PYGZhy{}out ../data/Sorghumvcf/Chr10\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz
\end{sphinxVerbatim}


\subsection{VCF Configuration file}
\label{\detokenize{VCFHunter:vcf-configuration-file}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{}Create a new vcf configuration file and fill it with this information(SorghumVcf.conf)}

../data/Sorghumvcf/Chr01\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz.recode.vcf
../data/Sorghumvcf/Chr02\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz.recode.vcf
../data/Sorghumvcf/Chr03\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz.recode.vcf
../data/Sorghumvcf/Chr04\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz.recode.vcf
../data/Sorghumvcf/Chr05\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz.recode.vcf
../data/Sorghumvcf/Chr06\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz.recode.vcf
../data/Sorghumvcf/Chr07\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz.recode.vcf
../data/Sorghumvcf/Chr08\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz.recode.vcf
../data/Sorghumvcf/Chr09\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz.recode.vcf
../data/Sorghumvcf/Chr10\PYGZus{}DNAseq\PYGZus{}Filtered\PYGZus{}filt.vcf.gz.recode.vcf
\end{sphinxVerbatim}


\subsection{Create a Origin tab delimited file}
\label{\detokenize{VCFHunter:create-a-origin-tab-delimited-file}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{}Create a new Origin tab delimited file and fill it with this information (SorghumOrigin.tab)}

con\PYGZhy{}all AA
D2      BB
\end{sphinxVerbatim}


\subsection{Create a Color configuration file}
\label{\detokenize{VCFHunter:create-a-color-configuration-file}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{}Create a new color configuration file and fill it with this information (SorghumColor.conf)}

AA \PYG{l+m}{0} \PYG{l+m}{255} \PYG{l+m}{0}
BB \PYG{l+m}{255} \PYG{l+m}{0} \PYG{l+m}{0}
\end{sphinxVerbatim}


\subsection{Run vcf2allPropandCov python script}
\label{\detokenize{VCFHunter:run-vcf2allpropandcov-python-script}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{}Run python script}
python vcf2allPropAndCov.py \PYGZhy{}\PYGZhy{}conf ../data/config/SorghumVcf.conf \PYGZhy{}\PYGZhy{}origin ../data/config/SorghumOrigin.tab \PYGZhy{}\PYGZhy{}acc D2\PYGZus{}F2\PYGZus{}tt \PYGZhy{}\PYGZhy{}ploidy \PYG{l+m}{2} \PYGZhy{}\PYGZhy{}dcurve y \PYGZhy{}\PYGZhy{}col /data/config/SorghumColor.conf
\end{sphinxVerbatim}


\subsection{Plots}
\label{\detokenize{VCFHunter:plots}}
\begin{figure}[htbp]
\centering

\noindent\sphinxincludegraphics{{D2_F2_ttCov}.png}
\end{figure}

\begin{figure}[htbp]
\centering

\noindent\sphinxincludegraphics{{D2_F2_ttRatio}.png}
\end{figure}

\begin{figure}[htbp]
\centering

\noindent\sphinxincludegraphics{{sorghumtab}.png}
\end{figure}



\renewcommand{\indexname}{Index}
\printindex
\end{document}